### PR TITLE
[PRO-2330] Better error handling

### PR DESCRIPTION
**Pull request by vovateknoir**
* Checks for data integrity
* Actual filtering valid data
* Added command for local debugging
* (?) Removed (seem)unnecessary dependency

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,13 @@
   "private": true,
   "scripts": {
     "test": "jest",
-    "start": "yarn workspace debug start"
+    "start": "yarn workspace debug start",
+    "debug": "NAMESPACE=demonstrations nodemon --watch src -e html,js,yaml --exec 'rm -vfr ./node_modules/node-red-teknoir-configure-devices/src/* && cp -Rv ./src/* ./node_modules/node-red-teknoir-configure-devices/src && yarn install && yarn workspace debug start-local'"
   },
+  "files": [
```
Comment (vovateknoir) [inline package.json:11]: AFIU these are basically what is needed for a package.

File: package.json [hunk 2]
```diff
@@ -36,13 +41,13 @@
     "depd": "2.0.0",
     "js-yaml": "3.14.0",
     "kubernetes-client": "9.0.0",
-    "merge-deep": "3.0.3",
-    "node-red-teknoir-configure-devices": "github:teknoir/node-red-teknoir-configure-devices#main"
```
Comment (vovateknoir) [inline package.json:40]: Do we need this dependency?

File: src/configure-device.js [hunk 3]
```diff
@@ -12,10 +12,10 @@ module.exports = function (RED) {
     }
 
     function configureDevice(node, device, namespace, config) {
-        const newItems = JSON.parse(JSON.stringify(config));
+        let newItems = JSON.parse(JSON.stringify(config));
 
         // prevent handling manifests that are not describing deployments (e.g. ConfigMap)
-        newItems.filter(manifest => manifest.hasOwnProperty('kind') && manifest.kind === 'Deployment').map(manifest => {
+        newItems = newItems.filter(manifest => manifest && manifest.hasOwnProperty('kind') && manifest.kind === 'Deployment').map(manifest => {
```
Comment (vovateknoir) [inline src/configure-device.js:18]: Now items actually filtered. So, any malformed node won't be accepted.

File: package.json (hunk 2 above)
Comment (tekn0ir) [reply package.json:40]: Most likely not

**Review by tekn0ir — CHANGES_REQUESTED** [review]
Can you bump the version in the [package](https://github.com/teknoir/node-red-teknoir-configure-devices/blob/f9fac0fa11c05381a08eeafc3be61ac76e3e67e1/package.json#L3), and create a branch with the same name.

You can also bump the version in the teknoir plugin catalog: https://github.com/teknoir/devstudio-catalog/blob/cd9c3d4fb30fae201353fb1bd4f7102f63ae753d/catalogue.json#L99

And after that bump the version in the Cloud Devstudio: https://github.com/teknoir/devstudio-cloud/blob/273a88c0ea67fc27acf4456e78b7277b047af2c9/Dockerfile#L68

And after that you can test it by creating the failing Devstudio in `teknoir-retail`. Create it with the name `retail`, and use existing Volume.

**Comment by vovateknoir** [comment]
@tekn0ir Done

[DS link](https://teknoir.cloud/_/devstudio/teknoir-retail/retail/?ns=boxer-property#flow/4b8ce15cda6baa20)  

https://github.com/teknoir/devstudio-catalog/pull/8
https://github.com/teknoir/devstudio-cloud/pull/28

image

image

**Review by jeryini — APPROVED** [review]

**Comment by vovateknoir** [comment]
Merged from another branch (https://github.com/teknoir/node-red-teknoir-configure-devices/pull/5)

**Pull request closed without merging**